### PR TITLE
Exclude resources from docs search

### PR DESCRIPTION
**Pull request by disrupted**

### CODE REVIEW

File: docs/docs/resources/architecture/components-hierarchy.md [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
     KubernetesApp --> PipelineComponent
-    KafkaConnector --> PipelineComponent
     KafkaApp --> KubernetesApp
     StreamsApp --> KafkaApp
     ProducerApp --> KafkaApp
+    KafkaConnector --> PipelineComponent
     KafkaSourceConnector --> KafkaConnector
     KafkaSinkConnector --> KafkaConnector
-    
-    click KubernetesApp "../kubernetes-app"
```
Comment (disrupted) [inline docs/docs/resources/architecture/components-hierarchy.md:11]: does it have to be relative link? Ideally, I'd wanna do something like /user/core-concepts/... but it doesn't work and setting it to /kpops/user/core-concepts/... breaks the docs version history

Comment (sujuka99) [reply docs/docs/resources/architecture/components-hierarchy.md:11]: How does it break the version history?

Comment (raminqaf) [reply docs/docs/resources/architecture/components-hierarchy.md:11]: Why don't you want it as relative links?

Comment (disrupted) [reply docs/docs/resources/architecture/components-hierarchy.md:11]: They result in 404 currently

Comment (sujuka99) [reply docs/docs/resources/architecture/components-hierarchy.md:11]: When did they break?

Comment (disrupted) [reply docs/docs/resources/architecture/components-hierarchy.md:11]: I suppose during restructuring https://github.com/bakdata/kpops/pull/329

Comment (raminqaf) [reply docs/docs/resources/architecture/components-hierarchy.md:11]: I can't reproduce the 404 bug. When I go to this link and click the components it takes me to the correct page 🤔 
https://bakdata.github.io/kpops/2.0/user/core-concepts/components/overview/

Comment (disrupted) [reply docs/docs/resources/architecture/components-hierarchy.md:11]: it's from this page https://bakdata.github.io/kpops/2.0/resources/architecture/components-hierarchy

Comment (disrupted) [reply docs/docs/resources/architecture/components-hierarchy.md:11]: perhaps we should simply exclude resources from the search?

Comment (sujuka99) [reply docs/docs/resources/architecture/components-hierarchy.md:11]: Yeah, I am pretty sure that this is not supposed to be seen on its own.

**Review by sujuka99 — APPROVED** [review]

**Review by raminqaf — APPROVED** [review]

**Pull request merged**